### ISSUE TITLE
How to do checkSession without making session rolling?

### ISSUE THREAD

**Issue by benevbright** [auth0/nextjs-auth0#1458]
### Checklist

- [X] I have looked into the [Readme](https://github.com/auth0/nextjs-auth0#readme), [Examples](https://github.com/auth0/nextjs-auth0/blob/main/EXAMPLES.md), and [FAQ](https://github.com/auth0/nextjs-auth0/blob/main/FAQ.md) and have not found a suitable solution or answer.
- [X] I have looked into the [API documentation](https://auth0.github.io/nextjs-auth0/) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/nextjs-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Describe the problem you'd like to have solved

I think it's pretty common feature but I can't find simple solution, which is kicking out user automatically after `rollingDuration` has passed. 

the same question in auth0 community: https://community.auth0.com/t/checksession-counts-as-session-activity/27521

basically, when I use `checkSession` in a poller (e.g every 10 minutes), it makes session rolling and can't achieve the auto logout. (classic thing that your heartbeat keeps you awake)

### Describe the ideal solution

there should be an way to check session without making session rolling. 

### Alternatives and current workarounds

- Only way I can think of is checking `user`'s `updated_at` and calculate the session's time which is not ideal. 
- or call logout directly after some user inactive time passed, which would make logic fragile.

**Comment by benevbright** [comment]
https://github.com/auth0/nextjs-auth0/pull/1116 this PR implemented similar thing I need.

**Comment by benevbright** [comment]
ok. what I want can be done by disabling `autoSave`. explained here: https://github.com/auth0/nextjs-auth0/pull/1116#issuecomment-1473841313

@adamjmcgrath @aovens-quantifi Is it possible to do other way around as well with current implementation? 

Basically I like to enable `autoSave: true` and have one specific endpoint that can check session without extending session. (or option in `checkSession` call)

**Comment by benevbright** [comment]
ok, after investigating a bit, the best option seems to be 

- configure `autoSave: true`
- call `const { checkSession } = useUser(); checkSession()` in poller
- put `await touchSession(req, res);` in our API router or in the router middleware. 

let me know if there is even better way

**Comment by aovens-quantifi** [comment]
We deliberately chose to do it this way in order to have the smallest blast radius. This way your "don't touch the session" logic can be whatever you want it to be and the rest of the time you just call touchSession

**Comment by benevbright** [comment]
EDIT: I previously mentioned `autoSave: true`. it's a simply mistake. I meant `autoSave: false. 

@aovens-quantifi thanks for the comment 👍

**Comment by cjsauer** [comment]
Quick question related to this issue: I can't seem to find `touchSession` as part of the public API anymore using version `3.1.0`. Has it been deprecated in favor of something else?

**Comment by benevbright** [comment]
@cjsauer I'm also using 3.1.0. you can find it from `initAuth0()` in api router.

**Comment by cjsauer** [comment]
Ah okay. Is there a reason that it isn't available for managed instances? Seems like it used to be from what I can tell combing through commit history.

**Comment by benevbright** [comment]
@cjsauer ok. I took a look a bit. It seems it was never implemented for managed instances, which seems a simple mistake in the beginning. I think you can open PR to add it. 
```
export const touchSession: TouchSession = (...args) => getInstance().touchSession(...args);
```

> Seems like it used to be from what I can tell combing through commit history.

I saw the commit that you might refer to but it was just moving around things that are in `initAuth0()`

**Comment by benevbright** [comment]
opened PR: https://github.com/auth0/nextjs-auth0/pull/1461
